# Imports

In [1]:
import pandas as pd

# Load data

In [2]:
df = pd.read_csv('../data/ILEDA_anonymized.csv')
df

Institution                                  Course actor.id  \
0              UEF  Advanced Data Management Systems (THJ)    aeb0c   
1              UEF  Advanced Data Management Systems (THJ)    aeb0c   
2              UEF  Advanced Data Management Systems (THJ)    aeb0c   
3              UEF  Advanced Data Management Systems (THJ)    aeb0c   
4              UEF  Advanced Data Management Systems (THJ)    aeb0c   
...            ...                                     ...      ...   
306740         BMU                                   IT131    2db36   
306741         BMU                                   IT131    fc5fe   
306742         BMU                                   IT131    d0f73   
306743         BMU                                   IT131    dfea0   
306744         BMU                                   IT131    7f889   

                   timestamp                                  verb.id  \
0       2023-07-07T18:55:47Z      http://id.tincanapi.com/verb/viewed   
1       2023-07-07T15:21:18Z      http://id.tincanapi.com/verb/viewed   
2       2023-07-07T15:08:13Z      http://id.tincanapi.com/verb/viewed   
3       2023-07-07T15:08:10Z      http://id.tincanapi.com/verb/viewed   
4       2023-07-07T09:03:32Z      http://id.tincanapi.com/verb/viewed   
...                      ...                                      ...   
306740  2023-06-16T16:52:55Z    http://adlnet.gov/expapi/verbs/scored   
306741  2023-06-16T17:58:19Z    http://adlnet.gov/expapi/verbs/scored   
306742  2023-06-16T17:08:43Z    http://adlnet.gov/expapi/verbs/scored   
306743  2023-06-16T17:43:28Z    http://adlnet.gov/expapi/verbs/scored   
306744  2023-06-09T16:23:38Z  http://adlnet.gov/expapi/verbs/answered   

              verb.display                             object.definition.name  \
0                   viewed             Advanced Data Management Systems (THJ)   
1                   viewed                  Assignment Instructions: ER & EER   
2                   viewed                           Recording: Intro lecture   
3                   viewed             Advanced Data Management Systems (THJ)   
4                   viewed             Advanced Data Management Systems (THJ)   
...                    ...                                                ...   
306740  attained grade for  Lekcija 7 - IP adresiranje i sabnetovanje - Tr...   
306741  attained grade for  Lekcija 8 - Transportni sloj - Tradicionalna n...   
306742  attained grade for  Lekcija 9 - Principi rada TCP protokola - Trad...   
306743  attained grade for   IT131 - L04 (638138597640325722 - Tradicionalno)   
306744            answered                               IT131-T02-Pitanje 15   

                                   object.definition.type  \
0         http://id.tincanapi.com/activitytype/lms/course   
1                http://adlnet.gov/expapi/activities/link   
2                http://adlnet.gov/expapi/activities/link   
3         http://id.tincanapi.com/activitytype/lms/course   
4         http://id.tincanapi.com/activitytype/lms/course   
...                                                   ...   
306740     http://adlnet.gov/expapi/activities/assessment   
306741     http://adlnet.gov/expapi/activities/assessment   
306742     http://adlnet.gov/expapi/activities/assessment   
306743     http://adlnet.gov/expapi/activities/assessment   
306744  http://adlnet.gov/expapi/activities/cmi.intera...   

                                                  context  \
0                                              UEF Moodle   
1       UEF Moodle > Advanced Data Management Systems ...   
2       UEF Moodle > Advanced Data Management Systems ...   
3                                              UEF Moodle   
4                                              UEF Moodle   
...                                                   ...   
306740                                      LAMS > 220870   
306741                                      LAMS > 221218   
306742                

# Duplicate values

In [3]:
(df.shape)

(306745, 12)

In [4]:
df.shape[0] - df.drop_duplicates().shape[0]

7454

There are 7454 duplicate rows in the dataset, which we have to drop.

In [5]:
df.drop_duplicates(inplace=True)
df.shape

(299291, 12)

# Dataset statistics

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 299291 entries, 0 to 306739
Data columns (total 12 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Institution             299291 non-null  object 
 1   Course                  299291 non-null  object 
 2   actor.id                299291 non-null  object 
 3   timestamp               299291 non-null  object 
 4   verb.id                 299291 non-null  object 
 5   verb.display            299291 non-null  object 
 6   object.definition.name  299291 non-null  object 
 7   object.definition.type  299291 non-null  object 
 8   context                 299291 non-null  object 
 9   result.score.scaled     11558 non-null   float64
 10  result.success          299291 non-null  bool   
 11  result.completion       299291 non-null  bool   
dtypes: bool(2), float64(1), object(9)
memory usage: 25.7+ MB


In [7]:
df.describe(include='all')

Institution                                   Course actor.id  \
count       299291                                   299291   299291   
unique           4                                        8      829   
top            UEF  Ihminen ja vuorovaikutteinen teknologia    76d01   
freq        153290                                   134855     6255   
mean           NaN                                      NaN      NaN   
std            NaN                                      NaN      NaN   
min            NaN                                      NaN      NaN   
25%            NaN                                      NaN      NaN   
50%            NaN                                      NaN      NaN   
75%            NaN                                      NaN      NaN   
max            NaN                                      NaN      NaN   

                   timestamp                              verb.id  \
count                 299291                               299291   
unique                245548                                   10   
top     2023-06-04T23:09:00Z  http://id.tincanapi.com/verb/viewed   
freq                      67                               229721   
mean                     NaN                                  NaN   
std                      NaN                                  NaN   
min                      NaN                                  NaN   
25%                      NaN                                  NaN   
50%                      NaN                                  NaN   
75%                      NaN                                  NaN   
max                      NaN                                  NaN   

       verb.display                   object.definition.name  \
count        299291                                   299291   
unique           10                                     1549   
top          viewed  Ihminen ja vuorovaikutteinen teknologia   
freq         229721                                    34133   
mean            NaN                                      NaN   
std             NaN                                      NaN   
min             NaN                                      NaN   
25%             NaN                                      NaN   
50%             NaN                                      NaN   
75%             NaN                                      NaN   
max             NaN                                      NaN   

                                 object.definition.type  \
count                                            299291   
unique                                               15   
top     http://id.tincanapi.com/activitytype/lms/course   
freq                                             102682   
mean                                                NaN   
std                                                 NaN   
min                                                 NaN   
25%                                                 NaN   
50%                                                 NaN   
75%                                                 NaN   
max                                                 NaN   

                                                  context  \
count                                              299291   
unique                                                189   
top     UEF Moodle > Ihminen ja vuorovaikutteinen tekn...   
freq                                                45880   
mean                                                  NaN   
std                                                   NaN   
min                                                   NaN   
25%                                                   NaN   
50%                                                   NaN   
75%                                                   NaN   
max                                                   NaN   

        result.score.scaled result.success result.completion  
count           11558.00000         299

In [8]:
df.groupby(df.columns.tolist(), as_index=False).size()

Institution                        Course actor.id  \
0             BMU                         CS120    05aba   
1             BMU                         CS120    05aba   
2             BMU                         CS120    05aba   
3             BMU                         CS120    05aba   
4             BMU                         CS120    05aba   
...           ...                           ...      ...   
11553          UL  ARQUITECTURA DE COMPUTADORES    fa4d0   
11554          UL  ARQUITECTURA DE COMPUTADORES    fa4d0   
11555          UL  ARQUITECTURA DE COMPUTADORES    fa4d0   
11556          UL  ARQUITECTURA DE COMPUTADORES    fee26   
11557          UL  ARQUITECTURA DE COMPUTADORES    fee26   

                  timestamp                                   verb.id  \
0      2023-03-18T18:55:20Z   http://adlnet.gov/expapi/verbs/answered   
1      2023-03-18T18:55:20Z   http://adlnet.gov/expapi/verbs/answered   
2      2023-03-18T18:55:20Z   http://adlnet.gov/expapi/verbs/answered   
3      2023-03-18T18:55:20Z   http://adlnet.gov/expapi/verbs/answered   
4      2023-03-18T18:55:20Z   http://adlnet.gov/expapi/verbs/answered   
...                     ...                                       ...   
11553  2023-06-01T16:59:31Z  http://adlnet.gov/expapi/verbs/completed   
11554  2023-06-02T10:47:07Z     http://adlnet.gov/expapi/verbs/scored   
11555  2023-06-07T09:15:26Z  http://adlnet.gov/expapi/verbs/completed   
11556  2023-02-23T17:11:26Z  http://adlnet.gov/expapi/verbs/completed   
11557  2023-06-02T10:44:09Z     http://adlnet.gov/expapi/verbs/scored   

             verb.display                      object.definition.name  \
0                answered                        CS120-T01-Pitanje 20   
1                answered                        CS120-T01-Pitanje 25   
2                answered                        CS120-T01-Pitanje 28   
3                answered                        CS120-T01-Pitanje 39   
4                answered                         CS120-T01-Pitanje 4   
...                   ...                                         ...   
11553           completed                            Lab#6 Defense G4   
11554  attained grade for                           Lab Assignment #6   
11555           completed             Defence quizz lab Assignment #4   
11556           completed  Initial evaluation quizz lab assignment #1   
11557  attained grade for                           Lab Assignment #6   

                                  object.definition.type  \
0      http://adlnet.gov/expapi/activities/cmi.intera...   
1      http://adlnet.gov/expapi/activities/cmi.intera...   
2      http://adlnet.gov/expapi/activities/cmi.intera...   
3      http://adlnet.gov/expapi/activities/cmi.intera...   
4      http://adlnet.gov/expapi/activities/cmi.intera...   
...                                                  ...   
11553             http://xapi.jisc.ac.uk/activities/quiz   
11554     http://adlnet.gov/expapi/activities/assessment   
11555             http://xapi.jisc.ac.uk/activities/quiz   
11556             http://xapi.jisc.ac.uk/activities/quiz   
11557     http://adlnet.gov/expapi/activities/assessment   

                                                 context  result.score.scaled  \
0                     LAMS > test > Pitanje 20 > Attempt             0.000000   
1                     LAMS > test > Pitanje 25 > Attempt             1.000000   
2                     LAMS > test > Pitanje 28 > Attempt             1.000000   
3                     LAMS > test > Pitanje 39 > Attempt             1.000000   
4                      LAMS > test > Pitanje 4 > Attempt             1.000000   
...                                                  ...                  ...   
11553  New Site > ARQUITECTURA DE COMPUTADORES - Grad...             0.560000   
11554  New Site > ARQUITECTURA DE COMPUTADORES - Grad...             0.560000   
11555  New Site > ARQUITECTURA DE COMPUTADORES - Grad...             0.708333   


# Columns

## Verb

In [9]:
df['verb.id'].value_counts()

verb.id
http://id.tincanapi.com/verb/viewed           229721
http://adlnet.gov/expapi/verbs/answered        33546
http://adlnet.gov/expapi/verbs/completed       18224
http://activitystrea.ms/schema/1.0/receive      6582
http://activitystrea.ms/schema/1.0/start        4400
http://adlnet.gov/expapi/verbs/scored           4236
http://activitystrea.ms/schema/1.0/submit       1481
http://activitystrea.ms/schema/1.0/create        931
http://activitystrea.ms/schema/1.0/join          136
http://activitystrea.ms/schema/1.0/leave          34
Name: count, dtype: int64

In [10]:
df['verb.id'] = df['verb.id'].str.split('/').str[-1]

In [11]:
df

Institution                                  Course actor.id  \
0              UEF  Advanced Data Management Systems (THJ)    aeb0c   
1              UEF  Advanced Data Management Systems (THJ)    aeb0c   
2              UEF  Advanced Data Management Systems (THJ)    aeb0c   
3              UEF  Advanced Data Management Systems (THJ)    aeb0c   
4              UEF  Advanced Data Management Systems (THJ)    aeb0c   
...            ...                                     ...      ...   
306723         BMU                                   IT131    87441   
306724         BMU                                   IT131    1c33c   
306730         BMU                                   IT131    98607   
306734         BMU                                   IT131    98607   
306739         BMU                                   IT131    00c5e   

                   timestamp    verb.id verb.display  \
0       2023-07-07T18:55:47Z     viewed       viewed   
1       2023-07-07T15:21:18Z     viewed       viewed   
2       2023-07-07T15:08:13Z     viewed       viewed   
3       2023-07-07T15:08:10Z     viewed       viewed   
4       2023-07-07T09:03:32Z     viewed       viewed   
...                      ...        ...          ...   
306723  2023-06-11T22:57:27Z   answered     answered   
306724  2023-05-16T00:05:45Z   answered     answered   
306730  2023-05-12T22:10:50Z  completed    completed   
306734  2023-02-27T22:45:33Z  completed    completed   
306739  2023-06-01T21:23:16Z   answered     answered   

                                   object.definition.name  \
0                  Advanced Data Management Systems (THJ)   
1                       Assignment Instructions: ER & EER   
2                                Recording: Intro lecture   
3                  Advanced Data Management Systems (THJ)   
4                  Advanced Data Management Systems (THJ)   
...                                                   ...   
306723                               IT131-T05-Pitanje 13   
306724                               IT131-T03-Pitanje 15   
306730                                          IT131-T03   
306734  Lekcija 2 - Osnove umrežavanja - Tradicionalna...   
306739                               IT131-T04-Pitanje 20   

                                   object.definition.type  \
0         http://id.tincanapi.com/activitytype/lms/course   
1                http://adlnet.gov/expapi/activities/link   
2                http://adlnet.gov/expapi/activities/link   
3         http://id.tincanapi.com/activitytype/lms/course   
4         http://id.tincanapi.com/activitytype/lms/course   
...                                                   ...   
306723  http://adlnet.gov/expapi/activities/cmi.intera...   
306724  http://adlnet.gov/expapi/activities/cmi.intera...   
306730     http://adlnet.gov/expapi/activities/assessment   
306734    http://id.tincanapi.com/activitytype/lms/module   
306739  http://adlnet.gov/expapi/activities/cmi.intera...   

                                                  context  \
0                                              UEF Moodle   
1       UEF Moodle > Advanced Data Management Systems ...   
2       UEF Moodle > Advanced Data Management Systems ...   
3                                              UEF Moodle   
4                                              UEF Moodle   
...                                                   ...   
306723                 LAMS > test > Pitanje 13 > Attempt   
306724                 LAMS > test > Pitanje 15 > Attempt   
306730                                        LAMS > test   
306734                                      LAMS > 217371   
306739                 LAMS > test > Pitanje 20 > Attempt   

        result.score.scaled  result.success  result.completion  
0                       NaN           False              False  
1                       NaN           False              False  
2                       NaN           False              False  
3                     

In [12]:
all(df['verb.id'].str.split('/').str[-1] == df['verb.display'])

False

In [13]:
df[df['verb.id'].str.split('/').str[-1] != df['verb.display']][['verb.id', 'verb.display']]

verb.id        verb.display
74       scored  attained grade for
96       scored  attained grade for
97       scored  attained grade for
98       scored  attained grade for
99       scored  attained grade for
...         ...                 ...
306087    start             started
306278  receive            received
306292  receive            received
306398    start             started
306439    start             started

[17800 rows x 2 columns]

In [14]:
df[['verb.id', 'verb.display']].drop_duplicates()

verb.id        verb.display
0          viewed              viewed
29      completed           completed
74         scored  attained grade for
714        submit           submitted
742        create             created
763       receive            received
892      answered            answered
906         start             started
153743       join              joined
153749      leave                left

Each ID corresponds to a single display.

In [22]:
df.drop(['verb.display'], axis=1, inplace=True)

In [23]:
df

Institution                            Course actor.id  \
0              UEF  Advanced Data Management Systems    aeb0c   
1              UEF  Advanced Data Management Systems    aeb0c   
2              UEF  Advanced Data Management Systems    aeb0c   
3              UEF  Advanced Data Management Systems    aeb0c   
4              UEF  Advanced Data Management Systems    aeb0c   
...            ...                               ...      ...   
306723         BMU                 Computer Networks    87441   
306724         BMU                 Computer Networks    1c33c   
306730         BMU                 Computer Networks    98607   
306734         BMU                 Computer Networks    98607   
306739         BMU                 Computer Networks    00c5e   

                   timestamp    verb.id  \
0       2023-07-07T18:55:47Z     viewed   
1       2023-07-07T15:21:18Z     viewed   
2       2023-07-07T15:08:13Z     viewed   
3       2023-07-07T15:08:10Z     viewed   
4       2023-07-07T09:03:32Z     viewed   
...                      ...        ...   
306723  2023-06-11T22:57:27Z   answered   
306724  2023-05-16T00:05:45Z   answered   
306730  2023-05-12T22:10:50Z  completed   
306734  2023-02-27T22:45:33Z  completed   
306739  2023-06-01T21:23:16Z   answered   

                                   object.definition.name  \
0                  Advanced Data Management Systems (THJ)   
1                       Assignment Instructions: ER & EER   
2                                Recording: Intro lecture   
3                  Advanced Data Management Systems (THJ)   
4                  Advanced Data Management Systems (THJ)   
...                                                   ...   
306723                               IT131-T05-Pitanje 13   
306724                               IT131-T03-Pitanje 15   
306730                                          IT131-T03   
306734  Lekcija 2 - Osnove umrežavanja - Tradicionalna...   
306739                               IT131-T04-Pitanje 20   

                                   object.definition.type  \
0         http://id.tincanapi.com/activitytype/lms/course   
1                http://adlnet.gov/expapi/activities/link   
2                http://adlnet.gov/expapi/activities/link   
3         http://id.tincanapi.com/activitytype/lms/course   
4         http://id.tincanapi.com/activitytype/lms/course   
...                                                   ...   
306723  http://adlnet.gov/expapi/activities/cmi.intera...   
306724  http://adlnet.gov/expapi/activities/cmi.intera...   
306730     http://adlnet.gov/expapi/activities/assessment   
306734    http://id.tincanapi.com/activitytype/lms/module   
306739  http://adlnet.gov/expapi/activities/cmi.intera...   

                                                  context  \
0                                              UEF Moodle   
1       UEF Moodle > Advanced Data Management Systems ...   
2       UEF Moodle > Advanced Data Management Systems ...   
3                                              UEF Moodle   
4                                              UEF Moodle   
...                                                   ...   
306723                 LAMS > test > Pitanje 13 > Attempt   
306724                 LAMS > test > Pitanje 15 > Attempt   
306730                                        LAMS > test   
306734                                      LAMS > 217371   
306739                 LAMS > test > Pitanje 20 > Attempt   

        result.score.scaled  result.success  result.completion  
0                       NaN           False              False  
1                       NaN           False              False  
2                       NaN           False              False  
3                       NaN           False              False  
4                       NaN           False              False  
...                     ...             ...                ...  
306723                1.000            True            

## Course

In [16]:
df[['Course', 'Institution']].drop_duplicates()

Course Institution
0                  Advanced Data Management Systems (THJ)         UEF
18463             Ihminen ja vuorovaikutteinen teknologia         UEF
153572  Проектиране на човеко-машинен интерфейс, летен...          SU
169434      Електронно обучение, летен семестър 2022/2023          SU
196531                       ARQUITECTURA DE COMPUTADORES          UL
271398                                   APLICACIONES WEB          UL
292147                                              CS120         BMU
300976                                              IT131         BMU

In [17]:
df['Course'].value_counts()

Course
Ihminen ja vuorovaikutteinen teknologia                              134855
ARQUITECTURA DE COMPUTADORES                                          74112
Електронно обучение, летен семестър 2022/2023                         26545
APLICACIONES WEB                                                      20699
Advanced Data Management Systems (THJ)                                18435
Проектиране на човеко-машинен интерфейс, летен семестър 2022/2023     15774
CS120                                                                  5263
IT131                                                                  3608
Name: count, dtype: int64

In [18]:
courses_dict = {
    'Ihminen ja vuorovaikutteinen teknologia': 'Human Factors of Interactive Technology',
    'Advanced Data Management Systems (THJ)': 'Advanced Data Management Systems',
    'Проектиране на човеко-машинен интерфейс, летен семестър 2022/2023': 'Human-computer interaction',
    'Електронно обучение, летен семестър 2022/2023': 'e-Learning',
    'ARQUITECTURA DE COMPUTADORES': 'Computer Architecture',
    'APLICACIONES WEB': 'Web Applications',
    'CS120': 'Computer Organization',
    'IT131': 'Computer Networks'
}

Note: BMU course name reference [here](https://www.metropolitan.ac.rs/files/2022/10/oas-informacione-tehnologije-2022.pdf)

In [19]:
df.replace({'Course': courses_dict}, inplace=True)

In [20]:
df

Institution                            Course actor.id  \
0              UEF  Advanced Data Management Systems    aeb0c   
1              UEF  Advanced Data Management Systems    aeb0c   
2              UEF  Advanced Data Management Systems    aeb0c   
3              UEF  Advanced Data Management Systems    aeb0c   
4              UEF  Advanced Data Management Systems    aeb0c   
...            ...                               ...      ...   
306723         BMU                 Computer Networks    87441   
306724         BMU                 Computer Networks    1c33c   
306730         BMU                 Computer Networks    98607   
306734         BMU                 Computer Networks    98607   
306739         BMU                 Computer Networks    00c5e   

                   timestamp    verb.id verb.display  \
0       2023-07-07T18:55:47Z     viewed       viewed   
1       2023-07-07T15:21:18Z     viewed       viewed   
2       2023-07-07T15:08:13Z     viewed       viewed   
3       2023-07-07T15:08:10Z     viewed       viewed   
4       2023-07-07T09:03:32Z     viewed       viewed   
...                      ...        ...          ...   
306723  2023-06-11T22:57:27Z   answered     answered   
306724  2023-05-16T00:05:45Z   answered     answered   
306730  2023-05-12T22:10:50Z  completed    completed   
306734  2023-02-27T22:45:33Z  completed    completed   
306739  2023-06-01T21:23:16Z   answered     answered   

                                   object.definition.name  \
0                  Advanced Data Management Systems (THJ)   
1                       Assignment Instructions: ER & EER   
2                                Recording: Intro lecture   
3                  Advanced Data Management Systems (THJ)   
4                  Advanced Data Management Systems (THJ)   
...                                                   ...   
306723                               IT131-T05-Pitanje 13   
306724                               IT131-T03-Pitanje 15   
306730                                          IT131-T03   
306734  Lekcija 2 - Osnove umrežavanja - Tradicionalna...   
306739                               IT131-T04-Pitanje 20   

                                   object.definition.type  \
0         http://id.tincanapi.com/activitytype/lms/course   
1                http://adlnet.gov/expapi/activities/link   
2                http://adlnet.gov/expapi/activities/link   
3         http://id.tincanapi.com/activitytype/lms/course   
4         http://id.tincanapi.com/activitytype/lms/course   
...                                                   ...   
306723  http://adlnet.gov/expapi/activities/cmi.intera...   
306724  http://adlnet.gov/expapi/activities/cmi.intera...   
306730     http://adlnet.gov/expapi/activities/assessment   
306734    http://id.tincanapi.com/activitytype/lms/module   
306739  http://adlnet.gov/expapi/activities/cmi.intera...   

                                                  context  \
0                                              UEF Moodle   
1       UEF Moodle > Advanced Data Management Systems ...   
2       UEF Moodle > Advanced Data Management Systems ...   
3                                              UEF Moodle   
4                                              UEF Moodle   
...                                                   ...   
306723                 LAMS > test > Pitanje 13 > Attempt   
306724                 LAMS > test > Pitanje 15 > Attempt   
306730                                        LAMS > test   
306734                                      LAMS > 217371   
306739                 LAMS > test > Pitanje 20 > Attempt   

        result.score.scaled  result.success  result.completion  
0                       NaN           False              False  
1                       NaN           False              False  
2                       NaN           False              False  
3                       NaN           False              False  
4                       NaN  

## Add teaching type column

Specify "Project-based" / "Flipped classroom" teaching style

In [24]:
def get_teaching_type(course_name):
    if course_name in ['Computer Networks',
                       'Advanced Data Management Systems',
                        'e-Learning',
                        'Computer Architecture']:
        return 'Flipped classroom'
    else:
        return 'Project-based'

In [25]:
df['Teaching'] = df['Course'].apply(get_teaching_type)

In [26]:
df

Institution                            Course actor.id  \
0              UEF  Advanced Data Management Systems    aeb0c   
1              UEF  Advanced Data Management Systems    aeb0c   
2              UEF  Advanced Data Management Systems    aeb0c   
3              UEF  Advanced Data Management Systems    aeb0c   
4              UEF  Advanced Data Management Systems    aeb0c   
...            ...                               ...      ...   
306723         BMU                 Computer Networks    87441   
306724         BMU                 Computer Networks    1c33c   
306730         BMU                 Computer Networks    98607   
306734         BMU                 Computer Networks    98607   
306739         BMU                 Computer Networks    00c5e   

                   timestamp    verb.id  \
0       2023-07-07T18:55:47Z     viewed   
1       2023-07-07T15:21:18Z     viewed   
2       2023-07-07T15:08:13Z     viewed   
3       2023-07-07T15:08:10Z     viewed   
4       2023-07-07T09:03:32Z     viewed   
...                      ...        ...   
306723  2023-06-11T22:57:27Z   answered   
306724  2023-05-16T00:05:45Z   answered   
306730  2023-05-12T22:10:50Z  completed   
306734  2023-02-27T22:45:33Z  completed   
306739  2023-06-01T21:23:16Z   answered   

                                   object.definition.name  \
0                  Advanced Data Management Systems (THJ)   
1                       Assignment Instructions: ER & EER   
2                                Recording: Intro lecture   
3                  Advanced Data Management Systems (THJ)   
4                  Advanced Data Management Systems (THJ)   
...                                                   ...   
306723                               IT131-T05-Pitanje 13   
306724                               IT131-T03-Pitanje 15   
306730                                          IT131-T03   
306734  Lekcija 2 - Osnove umrežavanja - Tradicionalna...   
306739                               IT131-T04-Pitanje 20   

                                   object.definition.type  \
0         http://id.tincanapi.com/activitytype/lms/course   
1                http://adlnet.gov/expapi/activities/link   
2                http://adlnet.gov/expapi/activities/link   
3         http://id.tincanapi.com/activitytype/lms/course   
4         http://id.tincanapi.com/activitytype/lms/course   
...                                                   ...   
306723  http://adlnet.gov/expapi/activities/cmi.intera...   
306724  http://adlnet.gov/expapi/activities/cmi.intera...   
306730     http://adlnet.gov/expapi/activities/assessment   
306734    http://id.tincanapi.com/activitytype/lms/module   
306739  http://adlnet.gov/expapi/activities/cmi.intera...   

                                                  context  \
0                                              UEF Moodle   
1       UEF Moodle > Advanced Data Management Systems ...   
2       UEF Moodle > Advanced Data Management Systems ...   
3                                              UEF Moodle   
4                                              UEF Moodle   
...                                                   ...   
306723                 LAMS > test > Pitanje 13 > Attempt   
306724                 LAMS > test > Pitanje 15 > Attempt   
306730                                        LAMS > test   
306734                                      LAMS > 217371   
306739                 LAMS > test > Pitanje 20 > Attempt   

        result.score.scaled  result.success  result.completion  \
0                       NaN           False              False   
1                       NaN           False              False   
2                       NaN           False              False   
3                       NaN           False              False   
4                       NaN           False              False   
...                     ...             ...                ...   
306723                1.000            True     

In [27]:
df['Teaching'].value_counts()

Teaching
Project-based        176591
Flipped classroom    122700
Name: count, dtype: int64

## Object

Trying machine translation

In [28]:
from easynmt import EasyNMT
model = EasyNMT("opus-mt")

ModuleNotFoundError: No module named 'easynmt'